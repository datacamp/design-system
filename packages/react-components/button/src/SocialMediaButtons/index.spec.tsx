import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import React from 'react';

import { FacebookButton, GoogleButton, LinkedinButton, TwitterButton } from '.';

jest.mock('react-uid', () => ({ useUIDSeed: () => () => 'mock-id' }));

describe('Social Media Buttons', () => {
  [
    ['facebook', FacebookButton],
    ['google', GoogleButton],
    ['linked in', LinkedinButton],
    ['twitter', TwitterButton],
  ].forEach(([name, Button]) => {
    it(`renders ${name} button`, async () => {
      const { container } = await axeRender(<Button />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
