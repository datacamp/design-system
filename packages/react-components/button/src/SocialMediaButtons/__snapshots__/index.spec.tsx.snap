// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Media Buttons renders facebook button 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  border-radius: 4px;
  border-style: solid;
  border-width: 2px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  vertical-align: baseline;
  white-space: nowrap;
  background-color: transparent;
  border-color: #05192d;
  color: #05192d;
  padding: 0 15px;
  border-color: #d9d9e2;
  color: #1778f2;
}

.emotion-0:active {
  -webkit-transform: perspective(1px) scale(0.975);
  -moz-transform: perspective(1px) scale(0.975);
  -ms-transform: perspective(1px) scale(0.975);
  transform: perspective(1px) scale(0.975);
}

.emotion-0:disabled,
.emotion-0:hover:disabled,
.emotion-0:active:disabled {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:hover {
  background-color: rgba(5, 25, 45, 0.15);
  border-color: #05192d;
  color: #05192d;
}

.emotion-0:hover {
  background-color: rgba(23, 120, 242, 0.15);
  border-color: #1778f2;
  color: #1778f2;
}

.emotion-1 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5, 25, 45);
  font-family: Studio-Feixen-Sans,Arial;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  font-size: 14px;
  line-height: 44px;
  color: #05192d;
  font-weight: bold;
  margin-left: 8px;
}

<a
  class="emotion-0"
>
  <svg
    aria-hidden="true"
    height="18"
    role="img"
    viewBox="0 0 18 18"
    width="18"
  >
    <path
      d="M18 9c0-4.968-4.032-9-9-9S0 4.032 0 9a8.99 8.99 0 007.6 8.889v-6.294H5.302V9H7.6V7.016c0-2.252 1.335-3.503 3.393-3.503.982 0 2.002.185 2.002.185v2.197h-1.13c-1.122 0-1.475.704-1.475 1.409V8.99h2.503l-.399 2.595H10.39v6.294A8.974 8.974 0 0018 9z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
  <span
    class="emotion-1"
  >
    Facebook
  </span>
</a>
`;

exports[`Social Media Buttons renders google button 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  border-radius: 4px;
  border-style: solid;
  border-width: 2px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  vertical-align: baseline;
  white-space: nowrap;
  background-color: transparent;
  border-color: #05192d;
  color: #05192d;
  padding: 0 15px;
  border-color: #d9d9e2;
  color: #db4437;
}

.emotion-0:active {
  -webkit-transform: perspective(1px) scale(0.975);
  -moz-transform: perspective(1px) scale(0.975);
  -ms-transform: perspective(1px) scale(0.975);
  transform: perspective(1px) scale(0.975);
}

.emotion-0:disabled,
.emotion-0:hover:disabled,
.emotion-0:active:disabled {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:hover {
  background-color: rgba(5, 25, 45, 0.15);
  border-color: #05192d;
  color: #05192d;
}

.emotion-0:hover {
  background-color: rgba(219, 68, 55, 0.15);
  border-color: #db4437;
  color: #db4437;
}

.emotion-1 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5, 25, 45);
  font-family: Studio-Feixen-Sans,Arial;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  font-size: 14px;
  line-height: 44px;
  color: #05192d;
  font-weight: bold;
  margin-left: 8px;
}

<a
  class="emotion-0"
>
  <svg
    aria-hidden="true"
    height="18"
    role="img"
    viewBox="0 0 18 18"
    width="18"
  >
    <g
      fill="none"
      fill-rule="nonzero"
    >
      <path
        d="M17.537 9.156c0-.59-.048-1.183-.15-1.763H9.205v3.34h4.685a4.015 4.015 0 01-1.734 2.636v2.168h2.796c1.641-1.511 2.585-3.742 2.585-6.381z"
        fill="#4285F4"
      />
      <path
        d="M9.205 17.63c2.34 0 4.312-.767 5.75-2.093l-2.795-2.168c-.778.53-1.782.83-2.952.83-2.263 0-4.182-1.528-4.87-3.58H1.453v2.234a8.676 8.676 0 007.752 4.778z"
        fill="#34A853"
      />
      <path
        d="M4.334 10.619a5.196 5.196 0 010-3.322V5.063H1.453a8.682 8.682 0 000 7.79l2.881-2.234z"
        fill="#FBBC04"
      />
      <path
        d="M9.205 3.715a4.714 4.714 0 013.327 1.3L15.01 2.54A8.337 8.337 0 009.205.282a8.673 8.673 0 00-7.752 4.781l2.881 2.234c.686-2.056 2.608-3.582 4.87-3.582z"
        fill="#EA4335"
      />
    </g>
  </svg>
  <span
    class="emotion-1"
  >
    Google
  </span>
</a>
`;

exports[`Social Media Buttons renders linked in button 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  border-radius: 4px;
  border-style: solid;
  border-width: 2px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  vertical-align: baseline;
  white-space: nowrap;
  background-color: transparent;
  border-color: #05192d;
  color: #05192d;
  padding: 0 15px;
  border-color: #d9d9e2;
  color: #0e76a8;
}

.emotion-0:active {
  -webkit-transform: perspective(1px) scale(0.975);
  -moz-transform: perspective(1px) scale(0.975);
  -ms-transform: perspective(1px) scale(0.975);
  transform: perspective(1px) scale(0.975);
}

.emotion-0:disabled,
.emotion-0:hover:disabled,
.emotion-0:active:disabled {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:hover {
  background-color: rgba(5, 25, 45, 0.15);
  border-color: #05192d;
  color: #05192d;
}

.emotion-0:hover {
  background-color: rgba(14, 118, 168, 0.15);
  border-color: #0e76a8;
  color: #0e76a8;
}

.emotion-1 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5, 25, 45);
  font-family: Studio-Feixen-Sans,Arial;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  font-size: 14px;
  line-height: 44px;
  color: #05192d;
  font-weight: bold;
  margin-left: 8px;
}

<a
  class="emotion-0"
>
  <svg
    aria-hidden="true"
    height="18"
    role="img"
    viewBox="0 0 18 18"
    width="18"
  >
    <path
      d="M4 2c0 1.1-.7 2-2 2-1.2 0-2-.9-2-1.9C0 1 .8 0 2 0s2 .9 2 2zM0 18h4V5H0v13zM13.6 5.2c-2.1 0-3.3 1.2-3.8 2h-.1l-.2-1.7H5.9c0 1.1.1 2.4.1 3.9V18h4v-7.1c0-.4 0-.7.1-1 .3-.7.8-1.6 1.9-1.6 1.4 0 2 1.2 2 2.8V18h4v-7.4c0-3.7-1.9-5.4-4.4-5.4z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
  <span
    class="emotion-1"
  >
    LinkedIn
  </span>
</a>
`;

exports[`Social Media Buttons renders twitter button 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  border-radius: 4px;
  border-style: solid;
  border-width: 2px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  vertical-align: baseline;
  white-space: nowrap;
  background-color: transparent;
  border-color: #05192d;
  color: #05192d;
  padding: 0 15px;
  border-color: #d9d9e2;
  color: #00acee;
}

.emotion-0:active {
  -webkit-transform: perspective(1px) scale(0.975);
  -moz-transform: perspective(1px) scale(0.975);
  -ms-transform: perspective(1px) scale(0.975);
  transform: perspective(1px) scale(0.975);
}

.emotion-0:disabled,
.emotion-0:hover:disabled,
.emotion-0:active:disabled {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-0:focus {
  outline: 0;
}

.emotion-0:hover {
  background-color: rgba(5, 25, 45, 0.15);
  border-color: #05192d;
  color: #05192d;
}

.emotion-0:hover {
  background-color: rgba(0, 172, 238, 0.15);
  border-color: #00acee;
  color: #00acee;
}

.emotion-1 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5, 25, 45);
  font-family: Studio-Feixen-Sans,Arial;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  font-size: 14px;
  line-height: 44px;
  color: #05192d;
  font-weight: bold;
  margin-left: 8px;
}

<a
  class="emotion-0"
>
  <svg
    aria-hidden="true"
    height="18"
    role="img"
    viewBox="0 0 18 18"
    width="18"
  >
    <path
      d="M15.864 5.35c.008.146.01.293.01.439 0 4.491-3.417 9.668-9.666 9.668A9.61 9.61 0 011 13.93c.266.031.536.047.81.047a6.819 6.819 0 004.22-1.453 3.4 3.4 0 01-3.174-2.36 3.472 3.472 0 001.534-.058 3.4 3.4 0 01-2.725-3.332V6.73a3.39 3.39 0 001.54.426A3.397 3.397 0 012.152 2.62a9.644 9.644 0 007.003 3.55A3.396 3.396 0 0112.466 2c.977 0 1.86.411 2.48 1.072a6.842 6.842 0 002.157-.825 3.41 3.41 0 01-1.494 1.88 6.796 6.796 0 001.951-.535 6.86 6.86 0 01-1.696 1.758z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
  <span
    class="emotion-1"
  >
    Twitter
  </span>
</a>
`;
